src/mainwin.cc

```diff
@@ -3047,9 +3097,21 @@ void MainWindow::consoleOutput(const QString &msg)
 	this->processEvents();
 }
 
+void MainWindow::errorLogOutput(const Message &log_msg, void *userdata)
+{
+	auto thisp = static_cast<MainWindow*>(userdata);
+	QMetaObject::invokeMethod(thisp, "errorLogOutput", Q_ARG(Message, log_msg));
+}
+
+void MainWindow::errorLogOutput(const Message &log_msg)
+{
+	this->errorLogWidget->toErrorLog(log_msg);
+}
+
 void MainWindow::setCurrentOutput()
 {
 	set_output_handler(&MainWindow::consoleOutput, this);
+	set_output_handler2(&MainWindow::errorLogOutput,this);
 }
 
 void MainWindow::hideCurrentOutput()
```

Comment:
Both `hideCurrentOutput()` and clearCurrentOutput()` need to do something with the 2nd handler too.

Reply:
Done